src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs

```diff
@@ -77,11 +77,10 @@ internal class HostListenerFactory : IListenerFactory
             return new CompositeListener(listeners);
         }
 
-        internal static bool IsDisabled(MethodInfo method, INameResolver nameResolver, IJobActivator activator)
+        internal static bool IsDisabled(MethodInfo method, INameResolver nameResolver, IJobActivator activator, IJobHostMetadataProvider metadataProvider)
```

Comment:
I think the layering here is backwards. If there's a [Disabled] attribute, IJobHostMetadataProvider.IsDisabled should also return true.

Reply:
although I see that creates a complicated bootstrapping problem in script. So maybe this is unavoidable. 

---
In reply to: [148617945](https://github.com/Azure/azure-webjobs-sdk/pull/1405#discussion_r148617945) [](ancestors = 148617945)

Reply:
Yeah I think layering is off here as well. We already have logic that computes disabled state and other per function metadata at index time. That's the same metadata that we should be returning via this new IJobHostMetadataProvider API.